drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -92,46 +93,27 @@ local function initialize_switch(device)
   -- is a workaround for those devices.
   if num_server_eps == 1 and detect_matter_thing(device) == true then
     device:try_update_metadata({profile = "switch-binary"})
-  elseif num_server_eps > 1 then
-    device:try_update_metadata({profile = string.format("switch-%d", math.min(num_server_eps, MAX_MULTI_SWITCH_EPS))})
-  end
-  if num_server_eps > MAX_MULTI_SWITCH_EPS then
-    error(string.format(
-      "Matter multi switch device will have limited function. Profile doesn't exist with %d components, max is %d",
-      num_server_eps,
-      MAX_MULTI_SWITCH_EPS
-    ))
   end
 end
 
 local function component_to_endpoint(device, component)
-  local map = device:get_field(COMPONENT_TO_ENDPOINT_MAP) or {}
```

Comment:
We talked about this offline too, but I think we need to keep the support for MCD devices in the driver since there may be existing devices relying on that functionality.